src/rubysys/array.rs

```diff
@@ -80,17 +80,13 @@ struct RArrayHeap {
     ptr: InternalValue,
 }
 
-#[repr(C)]
-struct RArray {
-    basic: RBasic,
-    as_: RArrayAs,
-}
+use rb_sys::RArray;
 
 pub unsafe fn rb_ary_len(value: Value) -> c_long {
     let rarray: *const RArray = mem::transmute(value.value);
     let flags = (*rarray).basic.flags;
 
-    if flags & (RArrayEmbed::Flag as size_t) == 0 {
+    if flags & (RArrayEmbed::Flag as u64) == 0 {
```

Comment:
```suggestion
    if flags & (RArrayEmbed::Flag as VALUE) == 0 {
```

Reply:
I switched to `InternalValue` which is aliased to `rb_sys::VALUE`.